jgiven-core/src/test/java/com/tngtech/jgiven/impl/inject/ValueInjectorTest.java

```diff
@@ -0,0 +1,43 @@
+package com.tngtech.jgiven.impl.inject;
+
+import com.tngtech.jgiven.annotation.ProvidedScenarioState;
+import com.tngtech.jgiven.annotation.ScenarioState;
+import com.tngtech.jgiven.exception.JGivenMissingGuaranteedScenarioStateException;
+import org.junit.Test;
+
+public class ValueInjectorTest {
+    private ValueInjector injector = new ValueInjector();
```

Comment:
Btw. Is there really no Unit test for value injectors yet?  
It might be a good idea to use the opportunity to also assert the behavior of the required attribute / the injection mechanism in general.

Reply:
Added some tests for the required field in the fixing commit.

Reply:
Bonus, have one of the "does not throw anything" tests assert that you actually can inject & read values. (for instance, by using the injector to initialize a public field)